File: content/skills/agility/courses/src/main/java/org/jesse/game/content/skills/agility/draynorrooftop/RoughWall.java
package org.jesse.game.content.skills.agility.draynorrooftop;

import org.jesse.game.content.skills.agility.AgilityCourseObstacle;
import org.jesse.game.content.skills.agility.MarkOfGrace;
import org.jesse.game.world.entity.Location;
import org.jesse.game.world.entity.player.Player;
import org.jesse.game.obj.ids.ObjectId;
import org.jesse.game.world.object.WorldObject;

/**
 * @author Tommeh | 25 feb. 2018 : 20:15:55
 * @see <a href="https://www.rune-server.ee/members/tommeh/">Rune-Server profile</a>}
 */
public final class RoughWall extends AgilityCourseObstacle {
	
	private static final Location END_LOC = new Location(3102, 3279, 3);

	public RoughWall() {
		super(DraynorRooftopCourse.class, 1);
	}

	@Override
	public void startSuccess(final Player player, final WorldObject object) {
		MarkOfGrace.spawn(player, DraynorRooftopCourse.MARK_LOCATIONS, 40, 10);
		player.useStairs(828, END_LOC, 1, 1);
	}

	@Override
	public int getLevel(final WorldObject object) {
		return 10;
	}

	@Override
	public int[] getObjectIds() {
		return new int[] {ObjectId.ROUGH_WALL };
	}

	@Override
	public double getSuccessXp(final WorldObject object) {
		return 5;
	}

	@Override
	public int getDuration(final boolean success, final WorldObject object) {
		return 1;
	}

}

File: content/skills/agility/courses/src/main/kotlin/org/jesse/game/content/skills/agility/draynorrooftop/RoughWall.kt
package org.jesse.game.content.skills.agility.draynorrooftop

import org.jesse.game.content.skills.agility.AgilityCourseObstacle
import org.jesse.game.content.skills.agility.MarkOfGrace
import org.jesse.game.obj.ids.ROUGH_WALL
import org.jesse.game.world.entity.Location
import org.jesse.game.world.entity.player.Player
import org.jesse.game.world.`object`.WorldObject

class RoughWall : AgilityCourseObstacle(DraynorRooftopCourse::class.java, 1) {
    override fun startSuccess(player: Player, `object`: WorldObject?) {
        MarkOfGrace.spawn(player, DraynorRooftopCourse.MARK_LOCATIONS, 40, 10)
        player.useStairs(828, END_LOC, 1, 1)
    }

    override fun getLevel(`object`: WorldObject?): Int {
        return 10
    }

    override fun getObjectIds(): IntArray {
        return intArrayOf(ROUGH_WALL)
    }

    override fun getSuccessXp(`object`: WorldObject?): Double {
        return 5.0
    }

    override fun getDuration(success: Boolean, `object`: WorldObject?): Int {
        return 1
    }

    companion object {
        private val END_LOC = Location(3102, 3279, 3)
    }
}
